### PR TITLE
Fix reaper usersecret

### LINKED ISSUE #433: Reaper does not deploy when release name != cluster name
## Bug Report

**Describe the bug**
If the release name and the cluster name used are not the same reaper fails to get the correct JMX secret and cannot start.

**To Reproduce**
1. Deploy with a configuration like, where the cluster name is NOT the same as the release name

```
cassandra:
  clusterName: k8ssandra-cluster
  heap:
   size: 500M
   newGenSize: 256M
  datacenters:
  - name: dc1
    size: 1
kube-prometheus-stack:
  prometheus:
    enabled: true
    prometheusSpec:
      externalUrl: http://localhost:9090/prometheus
      routePrefix: /prometheus
    ingress:
      enabled: true
      paths:
        - /prometheus
  grafana:
    enabled: true
    ingress:
      enabled: true
      path: /grafana
    adminUser: admin
    adminPassword: admin123
    grafana.ini:
      server:
        root_url: http://localhost:3000/grafana
        serve_from_sub_path: true
stargate:
  enabled: true
  replicas: 1
  ingress:
    enabled: true
reaper:
  ingress:
    enabled: true
    host: repair.127.0.0.1.nip.io
```

☝️ Not all of that is necessary, just the clusterName, this is just what I was using, for sake of completely

```
helm install -f k8ssandra.values.yaml --set cassandra.version=3.11.10 k8ssandra ~/github/k8ssandra/k8ssandra/charts/k8ssandra
```

☝️ note that the release name is `k8ssandra` and the cluster name is `k8ssandra-cluster`

2. Check the reaper logs and find:

```
2021-02-25T12:30:44.830Z	ERROR	controller	Reconciler error	{"reconcilerGroup": "reaper.cassandra-reaper.io", "reconcilerKind": "Reaper", "controller": "reaper", "name": "k8ssandra-reaper", "namespace": "default", "error": "Secret \"k8ssandra-reaper-jmx\" not found"}
```

**Expected behavior**
The reaper should use the appropriate secret

**Environment (please complete the following information):**

* Helm charts version info

```
% helm ls -a
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                   APP VERSION
k8ssandra       default         1               2021-02-25 08:07:02.116119 -0500 EST    deployed        k8ssandra-0.60.0        3.11.10    
traefik         default         1               2021-02-25 08:04:53.80431 -0500 EST     deployed        traefik-9.14.3          2.4.5
```

* Helm charts user-supplied values

```
% helm get values k8ssandra
USER-SUPPLIED VALUES:
cassandra:
  clusterName: k8ssandra-cluster
  datacenters:
  - name: dc1
    size: 1
  heap:
    newGenSize: 256M
    size: 500M
  version: 3.11.10
kube-prometheus-stack:
  grafana:
    adminPassword: admin123
    adminUser: admin
    enabled: true
    grafana.ini:
      server:
        root_url: http://localhost:3000/grafana
        serve_from_sub_path: true
    ingress:
      enabled: true
      path: /grafana
  prometheus:
    enabled: true
    ingress:
      enabled: true
      paths:
      - /prometheus
    prometheusSpec:
      externalUrl: http://localhost:9090/prometheus
      routePrefix: /prometheus
medusa:
  bucketName: jdinoto-k8ssandra-test
  enabled: true
  storage: s3
  storage_properties:
    region: us-east-1
  storageSecret: medusa-bucket-key
reaper:
  ingress:
    enabled: true
    host: repair.127.0.0.1.nip.io
stargate:
  enabled: true
  ingress:
    enabled: true
  replicas: 1
```

* Kubernetes version information:

```
% k version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.15", GitCommit:"2adc8d7091e89b6e3ca8d048140618ec89b39369", GitTreeState:"clean", BuildDate:"2021-01-22T22:12:48Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind:

Kind

**Additional context**

Was found testing #431

### PR DESCRIPTION
**Pull request by burmanm**
**What this PR does**:
Selects correct reaper JMX secret name as well as Cassandra secret name

**Which issue(s) this PR fixes**:
Fixes #433 

**Checklist**
- [ ] Changes manually tested
- [x] Chart versions updated (if necessary)
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: charts/k8ssandra/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: k8ssandra
 description: |
   Provisions and configures an instance of the entire K8ssandra stack. This includes Apache Cassandra, Stargate, Reaper, Medusa, Prometheus, and Grafana.
 type: application
-version: 0.60.0
+version: 0.60.1
```
Comment (jdonenine) [inline charts/k8ssandra/Chart.yaml:6]: I don't know why GH keeps failing to show the right diffs for the chart versions, same thing happened the other day.  In any case, we'll need to bump here, 60.1 is already current:

https://github.com/k8ssandra/k8ssandra/blob/ecdc4f1d78422b91c6ab4045e776314286cf8834/charts/k8ssandra/Chart.yaml#L6

File: charts/k8ssandra/templates/reaper/reaper.yaml [hunk 2]
```diff
@@ -10,13 +10,11 @@ spec:
...
-    jmxUserSecretName: {{ .Release.Name }}-reaper-jmx
+    jmxUserSecretName: {{ include "k8ssandra.reaperJmxUserSecretName" . }}
     cassandraBackend:
-    {{- if .Values.reaper.cassandraUser.secret }}
-      cassandraUserSecretName: {{ .Values.reaper.cassandraUser.secret }}
-    {{- else if or .Values.reaper.cassandraUser.username .Values.cassandra.auth.enabled }}
-      cassandraUserSecretName: {{ include "k8ssandra.clusterName" . }}-reaper
-    {{- end }}
+      {{- if .Values.cassandra.auth.enabled }}
+      cassandraUserSecretName: {{ default (include "k8ssandra.reaperUserSecretName" .) .Values.reaper.cassandraUser.secret }}
```
Comment (jsanda) [inline charts/k8ssandra/templates/reaper/reaper.yaml:16]: Can you refactor the `reaperUserSecretName` template to be like the `reaperJmxUserSecretName` template? Then this line can be simplified to:

```
cassandraUserSecretName: {{ include "k8ssandra.reaperUserSecretName" . }}
```

We should do the same for the other `k8ssandra.<component>UserSecretName` templates as well. We do this part in a separate, follow up PR though if you prefer.

**Review by jsanda — CHANGES_REQUESTED** [review]

File: charts/k8ssandra/templates/reaper/reaper.yaml (hunk 2 above)
Comment (burmanm) [reply charts/k8ssandra/templates/reaper/reaper.yaml:16]: It's not used just here. The behavior was the same in:

https://github.com/k8ssandra/k8ssandra/blob/158a80a154d0e1eee5d0558bc0e1c9fe6166c4e9/charts/k8ssandra/templates/cassandra/cassdc.yaml#L70

And that applies to almost every Cassandra username override

Comment (jsanda) [reply charts/k8ssandra/templates/reaper/reaper.yaml:16]: I will do a follow up PR for those changes.

**Review by jsanda — APPROVED** [review]

**Pull request merged**